navigation/ridf_processors.h

```diff
@@ -0,0 +1,115 @@
+#pragma once
+
+// Standard C++ includes
+#include <algorithm>
+#include <array>
+#include <limits>
+#include <numeric>
+#include <tuple>
+#include <vector>
+
+// OpenCV
+#include <opencv2/opencv.hpp>
+
+// BoB robotics
+#include "common/rtransform.h"
+
+// Third-party includes
+#include "../third_party/units.h"
+
+namespace BoBRobotics {
+namespace Navigation {
+using namespace units::angle;
+
+namespace Internal {
+template<typename T1, typename T2>
+static radian_t
+circularMean(const T1 &angles, const T2 &weights)
+{
+    scalar_t sumCos = 0.0;
+    scalar_t sumSin = 0.0;
+    for (size_t i = 0; i < angles.size(); i++) {
+        sumCos += weights[i] * units::math::cos(angles[i]);
+        sumSin += weights[i] * units::math::sin(angles[i]);
+    }
+
+    return units::math::atan2(sumSin / angles.size(), sumCos / angles.size());
+}
+}
+
+//! Winner-take all: derive heading using only the best-matching snapshot
+struct BestMatchingSnapshot
```

Comment:
I tried doing this with lambdas, but for some reason couldn't get it to work. You can't do:
```
template<typename someLambda>
...
```
because the compiler complains that a function is not a type, which I suppose makes sense.

We could presumably use:
```
template<std::function<...> someLambda
...
```
but I was too lazy to type this out the first time.

I'll have a go at seeing if I can do this though.

Reply:
I literally meant ye 'olde functions!

Reply:
Ohh... Yeah I did try that at first and it's the same deal as lambdas -- they're not "types". I'll try doing it with a ``std::function<...>`` thing instead though

Reply:
Ahh yes that does make sense - I feel maybe you can only supply actual functions to templated **functions**. Don't use std::function for this - it's runtime so **way** slower!